oncall/singlerulecalculator.go

```diff
@@ -0,0 +1,94 @@
+package oncall
+
+import "time"
+
+// SingleRuleCalculator will calculate the currently active user.
+type SingleRuleCalculator struct {
+	*TimeIterator
+
+	act     *ActiveCalculator
+	rot     *UserCalculator
+	loc     *time.Location
+	rule    ResolvedRule
+	userID  string
+	changed bool
+}
+
+// NewSingleRuleCalculator will create a new SingleRuleCalculator bound to the TimeIterator.
+func (t *TimeIterator) NewSingleRuleCalculator(loc *time.Location, rule ResolvedRule) *SingleRuleCalculator {
+	calc := &SingleRuleCalculator{
+		TimeIterator: t,
+		rule:         rule,
+		loc:          loc,
+		act:          t.NewActiveCalculator(),
+	}
+
+	loopLimit := 100000
```

Comment:
I think that would just be unnecessary indirection -- we only use it in this spot (within this package) -- if someone has to debug the resulting panic in the future, the stack will lead them here and they will have the context and the value present, rather than having to look for the const definition.